Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -237,30 +237,38 @@ def self.audit
     cask_results = if audit_casks.empty?
       {}
     else
-      require "cask/cmd/abstract_command"
-      require "cask/cmd/audit"
 
       if args.display_failures_only?
         odeprecated "`brew audit <cask> --display-failures-only`", "`brew audit <cask>` without the argument"
       end
 
-      # For switches, we add `|| nil` so that `nil` will be passed instead of `false` if they aren't set.
-      # This way, we can distinguish between "not set" and "set to false".
-      Cask::Cmd::Audit.audit_casks(
-        *audit_casks,
-        download:        nil,
-        # No need for `|| nil` for `--[no-]signing` because boolean switches are already `nil` if not passed
-        signing:         args.signing?,
-        online:          args.online? || nil,
-        strict:          args.strict? || nil,
-        new_cask:        args.new_cask? || nil,
-        token_conflicts: args.token_conflicts? || nil,
-        quarantine:      nil,
-        any_named_args:  !no_named_args,
-        language:        nil,
-        only:            args.only,
-        except:          args.except,
-      )
+      Homebrew.auditing = true
+
+      require "cask/auditor"
+
+      audit_casks.to_h do |cask|
+        odebug "Auditing Cask #{cask}"
+        [cask.sourcefile_path, { errors: Cask::Auditor.audit(
+          cask,
+          # For switches, we add `|| nil` so that `nil` will be passed
+          # instead of `false` if they aren't set.
+          # This way, we can distinguish between "not set" and "set to false".
+          audit_download:        nil,
+          audit_online:          (args.online? || nil),
+          audit_strict:          (args.strict? || nil),
+
+          # No need for `|| nil` for `--[no-]signing`
+          # because boolean switches are already `nil` if not passed
+          audit_signing:         args.signing?,
+          audit_new_cask:        (args.new_cask? || nil),
+          audit_token_conflicts: (args.token_conflicts? || nil),
+          quarantine:            true,
+          language:              nil,
+          any_named_args:        !no_named_args,
+          only:                  args.only,
+          except:                args.except,
+        ), warnings: [] }]
+      end
     end
 
     failed_casks = cask_results.reject { |_, result| result[:errors].empty? }
```

Comment:
```suggestion
    failed_casks = cask_results.reject { |_, errors| errors.empty? }
```

below here: all references to `result[:warnings]` can be removed and `result[: errors]` replaced with just `result`/`errors` because, if you apply suggestions above, `warnings` is never set/used and `errors` is the hash value.

Reply:
Hurray, this was a TODO that I wrote a note about in 8319c8f9b936cffa43571571e6eaf2f266f8e3df so it's great if this gets done.

Reply:
@hyuraku nudge on this!

Reply:
Opened https://github.com/Homebrew/brew/pull/15269 for this.